webapp/src/views/projects/translations/TranslationTools/MachineTranslation.tsx

```diff
@@ -36,76 +42,84 @@ const StyledValue = styled('div')`
   align-self: center;
 `;
 
+const StyledError = styled(StyledValue)`
+  color: ${({ theme }) => theme.palette.error.main};
+`;
+
 const StyledDescription = styled('div')`
   font-size: 13px;
   color: ${({ theme }) => theme.palette.text.secondary};
 `;
 
-type ProviderData = {
-  provider: string;
-  data: TranslationItemModel | undefined;
-};
-
 type Props = {
-  data: ProviderData[] | undefined;
+  machine: UseQueryResult<CombinedMTResponse, ApiError> | undefined;
   operationsRef: ReturnType<typeof useTranslationTools>['operationsRef'];
   languageTag: string;
   contextPresent: boolean | undefined;
-  isFetching: boolean;
 };
 
 export const MachineTranslation: React.FC<Props> = ({
-  data,
+  machine,
   operationsRef,
   languageTag,
   contextPresent,
-  isFetching,
 }) => {
   const { t } = useTranslate();
-  const baseIsEmpty = data?.find((item) => item.data === null);
-  const nothingFetched =
-    data?.every((item) => !item.data?.output) && isFetching;
+  const baseIsEmpty = false;
```

Comment:
Ah, yeah previously it was possible to find out that the base translation is empty, now I didn't find a reliable way. Maybe we could add it on backend?

Reply:
What to add on backend?

Reply:
When the base translation is empty it doesn't make sense to run the machine translation. Previously I was somehow detecting that if all the providers returned null and I've displayed a message `Base translation is empty`. Instead of empty results.

But it doesn't work now as it can be null also in other cases, so I think it might be better to indicate this from backend explicitly.